src/Entity/User.php

```diff
@@ -10,16 +10,12 @@
 class User implements UserInterface, PasswordAuthenticatedUserInterface
 {
     private int $id;
-    private string $email;
     private ?string $password;
-    private array $roles = [];
 
     public function __construct(
-        string $email,
-        array $roles = []
+        private readonly string $email,
+        private readonly array $roles = []
```

Comment:
I don't think it's a good idea to add readonly flag on email and roles 😉
```suggestion
        private string $email,
        private array $roles = []
```